temporal/api/cloud/cloudservice/v1/request_response.proto

```diff
@@ -229,3 +230,187 @@ message GetRegionResponse {
+    string task_queue = 4;
+
+    // Filter incoming services by their name - optional. Specifying this will result in zero or one results.
+    string name = 5;
+}
+
+message GetIncomingServicesResponse {
+    // The list of incoming services in ascending ids order
+    repeated temporal.api.cloud.nexus.v1.IncomingService incoming_services = 1;
+
+    // The next page's token
+    string next_page_token = 2;
+}
+
+message GetIncomingServiceRequest {
+    // The id of the incoming service to get
+    string incoming_service_id = 1;
+}
+
+message GetIncomingServiceResponse {
+    // The incoming service
+    temporal.api.cloud.nexus.v1.IncomingService incoming_service = 1;
+}
+
+message CreateIncomingServiceRequest {
+    // The spec for the incoming service
+    temporal.api.cloud.nexus.v1.IncomingServiceSpec spec = 1;
+
+    // The id to use for this async operation - optional
+    string async_operation_id = 2;
+}
+
+message CreateIncomingServiceResponse {
+    // The id of the service that was created
+    string incoming_service_id = 1;
+
+    // The async operation
+    temporal.api.cloud.operation.v1.AsyncOperation async_operation = 2;
+}
+
+message UpdateIncomingServiceRequest {
+    // The id of the incoming service to update
+    string incoming_service_id = 1;
+
+    // The updated incoming service specification
+    temporal.api.cloud.nexus.v1.IncomingServiceSpec spec = 2;
+
+    // The version of the incoming service for which this update is intended for
+    // The latest version can be found in the GetIncomingService operation response
+    string resource_version = 3;
+
+    // The id to use for this async operation - optional
+    string async_operation_id = 4;
+}
+
+message UpdateIncomingServiceResponse {
+    // The async operation
+    temporal.api.cloud.operation.v1.AsyncOperation async_operation = 1;
+}
+
+message DeleteIncomingServiceRequest {
+    // The id of the incoming service to delete
+    string incoming_service_id = 1;
+
+    // The version of the incoming service for which this delete is intended for
+    // The latest version can be found in the GetIncomingService operation response
+    string resource_version = 2;
+
+    // The id to use for this async operation - optional
+    string async_operation_id = 3;
+}
+
+message DeleteIncomingServiceResponse {
+    // The async operation
+    temporal.api.cloud.operation.v1.AsyncOperation async_operation = 1;
+}
+
+message GetOutgoingServicesRequest {
+    // The requested size of the page to retrieve - optional.
+    // Cannot exceed 1000. Defaults to 100. 
+    int32 page_size = 1;
+
+    // The page token if this is continuing from another response - optional.
+    string page_token = 2;
+
+    // The namespace to get all the outgoing services from
+    string namespace = 3;
+
+    // The name of the outgoing service to filter on - optional. Specifying this will result in zero or one results.
+    string name = 4;
```

Comment:
Not really useful as a filter, this is basically a Get request.

Reply:
right, this is what we did for our Users api (users have a static id, but you call GetUsersRequset and can supply an email address), so this is just an attempt to stay consistent.

Reply:
@anekkanti - thoughts? we basically followed the users model, which relies on this